packages/client/src/generation/TSClient/common.ts

```diff
@@ -524,7 +524,23 @@ type Cast<A, B> = A extends B ? A : B;
 
 export const type: unique symbol;
 
-export function validator<V>(): <S>(select: $Utils.LegacyExact<S, V>) => S;
+export function validator<V>():
```

Comment:
I think it should be fine to move it to runtime since it became way more complex than it used to be and also does not really depend on DMMF

Reply:
Agree, let's move as many parts as we can